src/txdb.cpp

```diff
@@ -274,7 +274,7 @@ bool CBlockTreeDB::LoadBlockIndexGuts(const Consensus::Params& consensusParams,
                 pindexNew->nStatus        = diskindex.nStatus;
                 pindexNew->nTx            = diskindex.nTx;
 
-                if (!CheckProofOfWork(pindexNew->GetBlockHash(), pindexNew->nBits, consensusParams))
+                if (pindexNew->GetBlockHash() != consensusParams.hashGenesisBlock && !CheckProofOfWork(pindexNew->GetBlockHash(), pindexNew->nBits, consensusParams))
```

Comment:
What happens if a node feeds us a nonsense header chain with millions of repeating "genesis blocks"? Is that caught in an earlier check (e.g. because `hashPrevBlock` points to a non-existing block)? cc @sdaftuar

Reply:
See #16551

Reply:
I think this problem can be generalized to "what does Bitcoin Core do if someone sends us the same block header repeatedly", in which case this change is no longer relevant, as the software *shouldn't* do a bunch of work to validate the same header over and over.